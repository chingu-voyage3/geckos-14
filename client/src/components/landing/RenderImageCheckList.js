import React from 'react';
import CheckListItem from './CheckListItem';

const RenderImageCheckList = props =>{
  const texts = props.texts;
  let checkList = [];
  texts.forEach(text => {
    checkList.push(<CheckListItem text={text} />);
  });
  return (
    <div className="features-one">
      <div>
        <h2>{props.heading}</h2>
      </div>
      <div className="features-one-container">
        <div>
           <img src={props.image} alt={props.alt} />
        </div>
        <div>
          {checkList}
        </div>
      </div>
    </div>
  );
};

export default RenderImageCheckList;
