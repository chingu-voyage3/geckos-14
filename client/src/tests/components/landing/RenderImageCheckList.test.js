import React from 'react';
import RenderImageCheckList from '../../../components/landing/RenderImageCheckList';
import board from '../../../assets/images/board.png';

import { shallow } from 'enzyme';

const checkListTexts = [
  'Discover all sensors/actuators connected to your device',
  'Add or remove readings from a specific sensor of a connected device',
  'Upload a file with sensor readings to display',
  'Discover all sensors/actuators connected to your Device'
];

it('renders without crashing', () => {
  const wrapper = shallow(<RenderImageCheckList
    heading="Features"
    image={board}
    alt = "A Web of things dashboard"
    texts = {checkListTexts}
   />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
