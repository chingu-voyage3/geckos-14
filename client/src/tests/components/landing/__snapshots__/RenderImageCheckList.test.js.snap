// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<section
  className="section"
>
  <h2
    className="text-center font-weight-bold"
  >
    Features
  </h2>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <img
        alt="A Web of things dashboard"
        className="img-fluid"
        src="board.png"
      />
    </div>
    <div
      className="col-md-6"
    >
      <CheckListItem
        key="Discover all sensors/actuators connected to your device"
        text="Discover all sensors/actuators connected to your device"
      />
      <CheckListItem
        key="Add or remove readings from a specific sensor of a connected device"
        text="Add or remove readings from a specific sensor of a connected device"
      />
      <CheckListItem
        key="Upload a file with sensor readings to display"
        text="Upload a file with sensor readings to display"
      />
      <CheckListItem
        key="Discover all sensors/actuators connected to your Device"
        text="Discover all sensors/actuators connected to your Device"
      />
    </div>
  </div>
</section>
`;
